src/ansys/dpf/composites/_indexer.py

```diff
@@ -252,39 +256,90 @@ def by_id(self, entity_id: int) -> Optional[NDArray[np.double]]:
         )
 
 
-class PropertyFieldIndexerWithDataPointerNoBoundsCheck:
-    """Indexer for a property field with data pointer and no bounds checks."""
+# DPF does not set the data pointers if a field has just
+# one value per entity. Therefore, it is unknown if
+# data pointer are set for some field of layered elements, for
+# instance angles, thickness etc.
+def get_field_indexer(field: Field) -> FieldIndexSingleValueProtocol:
```

Comment:
New function to get the correct FieldInder. This function and `get_property_field_indexer` should be used from externally only.